src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticToken.java

```diff
@@ -0,0 +1,281 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider;
+
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.ThreeState;
+import com.intellij.util.containers.ContainerUtil;
+import com.redhat.devtools.lsp4ij.features.semanticTokens.LSPSemanticTokenTypes;
+import org.eclipse.lsp4j.SemanticTokenModifiers;
+import org.eclipse.lsp4j.SemanticTokenTypes;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a concrete semantic token in a file.
+ */
+class LSPSemanticToken {
+
+    // All semantic token types
+    private static final Set<String> ALL_TOKEN_TYPES = ContainerUtil.union(
+            Set.of(
+                    SemanticTokenTypes.Namespace,
+                    SemanticTokenTypes.Type,
+                    SemanticTokenTypes.Class,
+                    SemanticTokenTypes.Enum,
+                    SemanticTokenTypes.Interface,
+                    SemanticTokenTypes.Struct,
+                    SemanticTokenTypes.TypeParameter,
+                    SemanticTokenTypes.Parameter,
+                    SemanticTokenTypes.Variable,
+                    SemanticTokenTypes.Property,
+                    SemanticTokenTypes.EnumMember,
+                    SemanticTokenTypes.Event,
+                    SemanticTokenTypes.Function,
+                    SemanticTokenTypes.Method,
+                    SemanticTokenTypes.Macro,
+                    SemanticTokenTypes.Modifier,
+                    SemanticTokenTypes.Comment,
+                    SemanticTokenTypes.String,
+                    SemanticTokenTypes.Number,
+                    SemanticTokenTypes.Regexp,
+                    SemanticTokenTypes.Operator,
+                    SemanticTokenTypes.Decorator
+            ),
+            Arrays.stream(LSPSemanticTokenTypes.values())
```

Comment:
This will be a recurring pattern where the standard semantic token types are merged with any custom ones. The notion of "custom" needs to be fleshed out more in response to #871.